datafusion/core/src/lib.rs

```diff
@@ -132,7 +132,13 @@
 //!
 //! ## Customization and Extension
 //!
-//! DataFusion supports extension at many points:
+//! DataFusion is designed to be a "disaggregated" query engine.  This
```

Comment:
This is trying to address @boazberman 's comments in https://github.com/apache/arrow-datafusion/discussions/6441#discussioncomment-6051065